src/base/coordinate-grid-mixin.ts

```diff
@@ -34,47 +35,35 @@ const DEFAULT_AXIS_LABEL_PADDING = 12;
  * @mixes MarginMixin
  */
 export class CoordinateGridMixin extends ColorMixin(MarginMixin) {
+    public _conf: ICoordinateGridMixinConf;
+
     private _parent: Selection<SVGElement, any, any, any>;
     private _g: SVGGElementSelection;
     private _chartBodyG: SVGGElementSelection;
     private _x: MinimalXYScale;
     private _origX: MinimalXYScale;
     private _xOriginalDomain: [number, number];
     private _xAxis: Axis<any>; // TODO: can we do better
-    private _xUnits: Units;
-    private _xAxisPadding: number;
-    private _xAxisPaddingUnit: string | CountableTimeInterval; // it can be string as well, like 'day', 'hour' etc.
-    private _xElasticity: boolean;
     private _xAxisLabel: string;
     private _xAxisLabelPadding: number;
     private _lastXDomain: [number, number];
     private _y: MinimalXYScale;
     private _yAxis: Axis<any>;  // TODO: can we do better
-    private _yAxisPadding: number;
-    private _yElasticity: boolean;
     private _yAxisLabel: string;
     private _yAxisLabelPadding: number;
     private _brush: BrushBehavior<unknown>;
     private _gBrush: SVGGElementSelection;
     private _brushOn: boolean;
```

Comment:
It looks somewhat arbitrary which members got moved to the config here.

I am clear that stuff that is generated by the chart and belongs to the chart state, e.g. the brush, zoom, and `_parentBrushOn`, stay in the chart.

But there are a log of user-configurable things which were left behind, like `x`/`yAxisLabel` & padding, the x and y scales. Does this have to do with side effects #1735 and internal modifications #1739?

Reply:
I think some cases are simple, where there is a simple side effect. I have experimented with SeriesChart in a similar case. See my comment in the respective commit.

In some cases where the properties are treated as an internal state, these must remain out of conf - for example, scales.

In no case modification of a conf property is allowed, as it will restrict sharing of conf across charts.

Few cases, like margins, are modified in complex ways in Coordinate charts, and even more complicated sharing through Composites. These cases would need refactoring before moving to conf. These would be taken on a case to case basis.

Reply:
Got it. Makes sense! 👍